include/drivers/bbram.h

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2021 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_BBRAM_H
+#define ZEPHYR_INCLUDE_DRIVERS_BBRAM_H
+
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * API template to check if the BBRAM is invalid.
+ *
+ * @see z_impl_bbram_get_ibbr
+ */
+typedef int (*bbram_api_get_ibbr)(const struct device *dev);
+
+/**
+ * API template to reset the invalid state.
+ *
+ * @see z_impl_bbram_reset_ibbr
+ */
+typedef int (*bbram_api_reset_ibbr)(const struct device *dev);
+
+/**
+ * API template to check for V SBY power failure.
+ *
+ * @see z_impl_bbram_get_vsby
+ */
+typedef int (*bbram_api_get_vsby)(const struct device *dev);
+
+/**
+ * API template to reset the V SBY status.
+ *
+ * @see z_impl_bbram_reset_vsby
+ */
+typedef int (*bbram_api_reset_vsby)(const struct device *dev);
+
+/**
+ * API template to check for V CC1 power failure.
+ *
+ * @see z_impl_bbram_get_vcc1
+ */
+typedef int (*bbram_api_get_vcc1)(const struct device *dev);
+
+/**
+ * API template to reset V CC1 status.
+ *
+ * @see z_impl_bbram_reset_vcc1
+ */
+typedef int (*bbram_api_reset_vcc1)(const struct device *dev);
+
+/**
+ * API template to read from BBRAM.
+ *
+ * @see z_impl_bbram_read
+ */
+typedef int (*bbram_api_read)(const struct device *dev, int offset, int size,
+			      uint8_t *data);
+
+/**
+ * API template to write to BBRAM.
+ *
+ * @see z_impl_bbram_write
+ */
+typedef int (*bbram_api_write)(const struct device *dev, int offset, int size,
+			       uint8_t *data);
+
+__subsystem struct bbram_driver_api {
+	bbram_api_get_ibbr get_ibbr;
+	bbram_api_reset_ibbr reset_ibbr;
+	bbram_api_get_vsby get_vsby;
+	bbram_api_reset_vsby reset_vsby;
+	bbram_api_get_vcc1 get_vcc1;
+	bbram_api_reset_vcc1 reset_vcc1;
+	bbram_api_read read;
+	bbram_api_write write;
+};
+
+__syscall int bbram_get_ibbr(const struct device *dev);
+
+/**
+ * Check if "Invalid Battery-Backed RAM". This may occur as a result to low
+ * voltage at the VBAT pin.
+ *
+ * @param dev BBRAM device pointer.
+ * @return 0 if the Battery-Backed RAM data is valid.
+ */
+static inline int z_impl_bbram_get_ibbr(const struct device *dev)
```

Comment:
I am thinking about this API in terms of Zephyr

bbr seems redundant 

I wonder if we should combine the get and reset?

Like bbram_check_inval() that checks if it is invalid, returns it if so, and returns 0 if the data is valid, -ENODATA if the data was not valid?

It doesn't seem that useful to request the info and not reset it...at least we don't in ECOS

Reply:
Done